internal/controllers/core/dockercomposeservice/reconciler.go

```diff
@@ -202,7 +212,12 @@ func (r *Reconciler) recordSpecAndDisableStatus(
 	defer r.mu.Unlock()
 
 	result := r.ensureResultExists(nn)
-	result.Spec = spec
+	if !apicmp.DeepEqual(result.Spec, spec) {
+		delete(r.resultsByServiceName, result.Spec.Service)
+		result.Spec = spec
+		result.ProjectHash, _ = hashProject(spec.Project)
```

Comment:
Worth making this `mustHashProject` and panic'ing if there's an error?

(afaict an error would mean an invalid assumption, e.g. unmarshalable project)